iothub/device/tests/TelemetryMethodsTests.cs

```diff
@@ -31,8 +33,25 @@ public void GetSqmMachineId_ReturnsExpectedValue()
             }
             else
             {
-                // GetSqmMachineId() should always return null for all other platforms
-                Assert.IsNull(actualValue);
+                if (File.Exists("/etc/machine-id"))
+                {
+                    // check if actualValue is a valid OSF v4 UUID
+                    // https://www.freedesktop.org/software/systemd/man/machine-id.html
```

Comment:
I don't follow why is this check a valid test for the SDK?

Reply:
The _SQMMachineId_ generated by the SDK needs to be a valid OSF v4 UUID. The machine-id [documentation](https://www.freedesktop.org/software/systemd/man/machine-id.html#Relation%20to%20OSF%20UUIDs) calls out the bitwise operations that decode the ID into a valid OSF v4 UUID.
This is what is being tested here - if the _SQMMachineId_ generated by the SDK has the right bits set.